libdnf5/base/transaction.cpp

```diff
@@ -158,6 +162,28 @@ static std::vector<std::pair<ProblemRules, std::vector<std::string>>> get_remova
     return problem_output;
 }
 
+class TransactionLocker : public libdnf5::utils::Locker {
+public:
+    TransactionLocker(const std::filesystem::path & lock_path, const ActiveTransactionInfo & info);
+
+    void write_transaction_metadata();
+
+private:
+    ActiveTransactionInfo transaction_info;
+};
+
+TransactionLocker::TransactionLocker(const std::filesystem::path & path, const ActiveTransactionInfo & info)
+    : Locker(path.string()),
+      transaction_info(info) {}
+
+void TransactionLocker::write_transaction_metadata() {
+    try {
+        write_content(transaction_info.to_toml());
+    } catch (const SystemError & e) {
+        //TODO(mblaha): log the error (would require base object)
```

Comment:
At least in the interim should this just go to stderr?

Reply:
Hm, we try to keep libdnf5 quiet. It does not print to stderr at all.

Reply:
If libdnf5 is logging, stderr is not all that different, right?  If we are to log here but are not [yet], then stderr feels appropriate.  Either that or raise an exception.